pkg/controllers/management/auth/auth_config.go

```diff
@@ -125,12 +137,26 @@ func (ac *authConfigController) sync(key string, obj *v3.AuthConfig) (runtime.Ob
 
 		switch value {
 		case CleanupUnlocked:
-			err := ac.cleanup.Run(obj)
+			// First, reset the auth config by removing all but essential metadata fields.
+			u, err := ac.getUnstructured(obj)
+			if err != nil {
+				return nil, err
+			}
+			cfg := u.UnstructuredContent()
+			common.ResetAuthConfig(cfg)
+			_, err = ac.authConfigsUnstructured.Update(obj.Name, u)
+			if err != nil {
+				return nil, fmt.Errorf("failed to update AuthConfig object: %w", err)
+			}
+
+			// Second, run resource cleanup.
+			err = ac.cleanup.Run(obj)
 			if err != nil {
 				return obj, err
 			}
+
+			// Third, lock the config after cleanup.
 			logrus.Infof("Auth provider %s has been cleaned up successfully. Locking down its cleanup operation...", obj.Name)
-			// Lock the config after cleanup.
 			return ac.setCleanupAnnotation(obj, CleanupRancherLocked)
```

Comment:
At this point the auth config will be updated by L147 but `obj` will not have been updated. `setCleanupAnnotation` also calls update on `obj`, but it will be out of date and cause a conflict error. At minimum, line 147 should return the object so that `setCleanupAnnotation` has an up to date object to work with. Better would be to refactor this so that the reset and annotation are made to the object and then call Update only once on the final object.